gcivil-nyu-org/fall2019-cs-gy-6063-team-debug_entities
# 61 implement login checks

I changed our user authentication to use a library named [allauth](https://django-allauth.readthedocs.io/en/latest/). The benefits of using allauth are:
* Built-in email verification tools
* We no longer have to manually manage every aspect of user creation and verification
* We won't have to write code to hash or verify passwords

How to test this PR:
* Get the code from this PR
* Run `pip install django-allauth`
* Delete migrations files and db.sqlite3
* Run makemigrations and migrate
* Run the server and open the page. Click "Sign Up" and fill in the form.
* See that you receive an email. Don't click the link yet.
* Try to sign in. It fails, as desired, because you haven't verified your email.
* Click the link in the email. Try to log in again but use an incorrect password. It fails, as desired.
* Log in with your correct password. See that it works.
* Optional: run `python manage.py pull_seatgeek_data` and then go to the events/ URL to see that the events are shown. The changes in this PR should not have any effect on the events page.
